app/items/[uuid]/page.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Metadata } from "next";
+import PageLayout from "app/components/pageLayout/pageLayout";
+import Item from "@/components/items/item";
+import { getItemData } from "@/utils/utils";
+import { ItemModel } from "@/models/item";
+
+type ItemProps = {
+  params: {
+    uuid: string;
+    item: ItemModel;
+  };
+};
+
+const getItem = async (uuid) => {
+  const item = await getItemData(uuid);
+  return item;
+};
+
+export async function generateMetadata({
+  params,
+}: ItemProps): Promise<Metadata> {
+  const data = await getItem(params.uuid);
+  const item = new ItemModel(data);
```

Comment:
I think it's okay to move this line inside of the `getItem` function and return it. The `getItemData` function itself is already doing the fetching so `getItem` is really a wrapper but not doing anything else. So you could have done, with the same result, 
```
const data = await getItemData(params.uuid);
const item = new ItemModal(data);
```

If you do move line 24 inside of `getItem`, it could be renamed to reflect what it's doing such as `getItemModel`.